### DISCUSSION TITLE
Minor issues and requests

### DISCUSSION THREAD

**Discussion by janert** [bbidulock/icewm#674] [General]
First of, congrats to the release of IceWM 3.0! I haven't had a chance to check it out yet, but it's great news!

In the meantime, I wanted to mention a few minor issues and requests:

1) In version 2.9.6 (which is what comes with Mint 21 Vanessa), when an application opens a "large" window, then the titlebar of the window is placed ABOVE the physical screen, making it impossible to move the window with the mouse. (It still can be moved using Alt-F7). That seems to be a regression; I don't recall ever having this problem before. (It seems as if EXACTLY the titlebar is cut off, suggesting that its height is not taken into account when placing the window.)

2) I noticed that the icons on the buttons in the taskbar sometimes change. For example, when I have several Emacs windows open, then some of the taskbar buttons show different Emacs icons. (See attachment.) Moreover, the icons sometimes change around  dynamically: the attachment is just a momentary snapshot. I have not been able to make out a pattern when and how they change (i.e. when I open and close windows, when I load docs into Emacs). Not a major issue, but weird.
![taskbar](https://user-images.githubusercontent.com/5132066/195214311-f693f53b-cb84-4d85-835a-8285f23c39a3.png)

3) Feature request: It's possible to run an application by clicking the clock, CPU, and network monitors in the taskbar, but it does NOT seem to be possible to do so on the memory monitor. That's too bad; it would be useful to be able to do so. (I looked at the code briefly, and there does seem to be a handler for this feature, but it doesn't seem to be active.) Would be great to activate that.

4) Documentation request: When creating a Theme, the "Look" parameter can have a pretty big influence on how IceWM appears, but the details don't seem to be documented anywhere. It took me quite a bit of confusion and trial-and-error to figure out what the "Look" does. For example, only with "gtk" and "pixmap" do I get the "sunken" look for the taskbar button of the active window. Only with "motif" (I believe) do the window frames indicate the extent of the corner size-grips. Only "metal" (and "motif"?) draw a frame around taskbar buttons. And so on. It would be great to document at least some of these details in the (otherwise terrific!) Theme HowTo.

**Comment by gijsbers** [comment]
1. Define "large".
2. To see what causes this: icewm can trace icon loading. Observe the properties of a window with xprop.
3. What application is relevant?
4. That is a lot of tedious work. Consider donating. Or you can do it. You have the motivation.

**Reply by janert** [reply]
1) Window Height >= Screen Height. The problem is not that the window doesn't fit, but that the titlebar is inaccessible.
2) Let me try and report back if I find something. It's fiddly, because the problem is intermittent and non-predicable (from the outside).
3) Click clock: show calendar. Click CPU: show top. Click network: show netstat. => Click memory: show vmstat. (along those lines)
4) I knew you were going to say that! ;-) I may (or may not) have the motivation, but I don't have the knowledge. I thought that maybe someone from long ago still remembered most of what this parameter does and could jot it down. I didn't mean for anyone to go on a deep dive to trace down every last bit of behavior.

**Comment by gijsbers** [comment]
1. dunno what causes this. Maybe some like the new behavior?
3. is the output of vmstat sufficiently user-friendly to warrant the added options and documentation?
4. we could start a "Volunteer wanted" topic?

**Reply by janert** [reply]
1) Ok. May be application specific. Not a big deal.
2) pass
3) TLDR: Yes! If it makes sense to support this kind of "hook" for CPU and Net monitors, then it makes sense to be consistent and also support it for memory. It's also not about vmstat specifically: I can click the CPU and Net monitors to run a command that gives me more details on the CPU or network situation. Why should I not be able to do the same for the memory monitor? In a larger context: The taskbar monitors are one of my favorite aspects of IceWM: small, unobtrusive, but always visible to know what's happening. And IF something looks odd, with a single click I can get a detailed view what is consuming (say) all my CPU. It seems only natural to do the same with memory. 
4) Go for it... it's really a call to the larger, original developer community that might still remember what this parameter does, to capture that info while it is still available, if possible and desired. That's all. ;-)

**Comment by gijsbers** [comment]
1. still something to keep in back of mind.
2. .
3. I'm sensitive to consistency, but not to satisfy just one person. If a dozen others would raise their hand, OK, but otherwise it must be something more user-friendly than vmstat.
4. me? huh?

**Reply by janert** [reply]
Just on the memory monitor: This is really not about vmstat, but about the bare capability. Personally, I have custom scripts hooked up to the CPU and Net monitors, as well as to the clock. I have a similar custom script for memory stuff that I would like to hook up to the mem monitor in a similar fashion, rather than having to open a terminal separately and run it there. 

Another example for something that is not vmstat but would make sense on the Mem monitor is something like this: 
https://github.com/pixelb/ps_mem

I understand your point about not creating one-offs, but supporting capabilities for users to utilize in any way they want strikes me as goodness, quite generally, and very much in the iceWM spirit. Also, I never understood why, if Net, CPU, and Clock are instrumented this way, the Mem monitor isn't...

Surely, not the most urgent item, by far. Just a friendly suggestion to extend the existing functionality in a consistent fashion.
